packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -459,6 +463,10 @@ const DataGrid = injectIntl((props) => {
       event.preventDefault(); // prevent the page from moving with the arrow keys.
       return;
     }
+
+    const newRowId = nextRow === 0 ? undefined : dataGridRows[nextRow - 1]?.id;
+    setLastActiveRowId(newRowId);
```

Comment:
Shouldn't we be setting this value where we set the focused row and column?  I wouldn't expect them to be separated.